packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -29,8 +35,14 @@ Future<void> main(FutureOr<void> testMain()) async {
 /// files from a sibling repository, `flutter/goldens`.
 ///
 /// This comparator will locally clone the `flutter/goldens` repository into
-/// the `$FLUTTER_ROOT/bin/cache/pkg/goldens` folder, then perform the comparison against
-/// the files therein.
+/// the `$FLUTTER_ROOT/bin/cache/pkg/goldens` folder using the [GoldensClient],
+/// then perform the comparison against the files therein.
+///
+/// For testing across all platforms, the [SkiaGoldClient] is used to upload
+/// widgets for framework-related golden tests and process results. Currently
+/// these tests are designed to be run in CI post-submit.
+///
+/// This comparator will instantiate [GoldensClient] and [SkiaGoldClient].
```

Comment:
Combining both features(repo-based and sikia gold) into one class makes this kinda confusing and hard to follow what's going on. Would it maybe make sense to have two seperate classes implementing GoldenFileComparator, one FlutterGoldenRepositoryFileComparator and the other one FlutterSkiaGoldFileComparator and the main above just choses the right one based on the environment? (You may want to give those two a common base class if there is logic that's the same for both)

Reply:
I still feel like completely separating the two impls would be a good idea.

Reply:
+1 (if nothing else, it'd allow you to avoid the `dynamic` argument).  However, if you had a common interface between the two clients (see comments below), then it's possible you could share code in a base comparator for the common parts.